CPB_AMS_Gizmo_BLE/cpb_ams_gizmo_ble.py

```diff
@@ -0,0 +1,130 @@
+radio = adafruit_ble.BLERadio() # pylint: disable=no-member
+a = SolicitServicesAdvertisement()
+a.solicited_services.append(AppleMediaService)
+radio.start_advertising(a)
+
+def wrap_in_tilegrid(open_file):
+    odb = displayio.OnDiskBitmap(open_file)
+    return displayio.TileGrid(odb, pixel_shader=displayio.ColorConverter())
+
+def make_background(width, height, color):
+    color_bitmap = displayio.Bitmap(width, height, 1)
+    color_palette = displayio.Palette(1)
+    color_palette[0] = color
+
+    return displayio.TileGrid(color_bitmap,
+                              pixel_shader=color_palette,
+                              x=0, y=0)
+
+def load_font(fontname, text):
+    font = bitmap_font.load_font(fontname)
+    font.load_glyphs(text.encode('utf-8'))
+    return font
+
+def make_label(text, x, y, color, max_glyphs=30, font=terminalio.FONT):
+    if isinstance(font, str):
+        font = load_font(font, "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz.,?()")
+    text_area = Label(font, text=text, color=color, max_glyphs=max_glyphs)
+    text_area.x = x
+    text_area.y = y
+    return text_area
+
+def set_label(label, value, max_length):
+    text = "{}".format(value)
+    if len(text) > max_length:
+        text = text[:max_length-3] + "..."
+    label.text = text
+
+def set_status(label, action_text, player):
+    label.text = "{} on {}".format(action_text, player)
+    _, _, label_width, _ = label.bounding_box
+    label.x = display.width - 10 - label_width
+
+display = tft_gizmo.TFT_Gizmo()
+group = displayio.Group(max_size=20)
+display.show(group)
+
+while True:
+    if not radio.connected:
+        group.append(wrap_in_tilegrid(open("/graphic_tfts_ams_connect.bmp", "rb")))
+
+        while not radio.connected:
+            pass
+
+        group.pop()
+    print("connected")
+
+    known_notifications = set()
+
+    # Draw the text fields
+    print("Loading Font Glyphs...")
+    title_label = make_label("None", 18, 30, TEXT_COLOR, font="/fonts/Arial-Bold-24.bdf")
+    artist_label = make_label("None", 18, 70, TEXT_COLOR, font="/fonts/Arial-22.bdf")
+    album_label = make_label("None", 18, 170, TEXT_COLOR, font="/fonts/Arial-22.bdf")
+    status_label = make_label("None", 80, 220, STATUS_COLOR, font="/fonts/Arial-16.bdf")
+    group.append(make_background(240, 240, BACKGROUND_COLOR))
+    border = Rect(10, 8, 200, 190, outline=BORDER_COLOR, stroke=1)
+    group.append(title_label)
+    group.append(artist_label)
+    group.append(album_label)
+    group.append(status_label)
+    group.append(border)
+
+    while radio.connected:
+        try:
+            for connection in radio.connections:
+                if not connection.paired:
+                    connection.pair()
+                    print("paired")
+
+                ams = connection[AppleMediaService]
+                set_label(title_label, ams.title, 12)
+                set_label(album_label, ams.album, 13)
```

Comment:
increase max characters to fit w more space, smaller font

Reply:
same

Reply:
Looks terrific!